File: medical_qa_app/test/chat_attachment_retention_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:medical_qa_app/utils/chat_attachment_retention.dart';

void main() {
  group('ChatAttachmentRetention', () {
    test('uses a 180 day retention window', () {
      expect(ChatAttachmentRetention.retentionDays, 180);
    });

    test('expires chat attachments when the message is older than retention',
        () {
      final now = DateTime.utc(2026, 7, 1);

      expect(
        ChatAttachmentRetention.isExpired(
          DateTime.utc(2026, 1, 1),
          now: now,
        ),
        isTrue,
      );
      expect(
        ChatAttachmentRetention.isExpired(
          DateTime.utc(2026, 6, 1),
          now: now,
        ),
        isFalse,
      );
    });
  });
}

File: admin-web/src/utils/__tests__/chatAttachmentRetention.test.js
import {
  CHAT_ATTACHMENT_RETENTION_DAYS,
  isChatAttachmentExpired,
} from '../chatAttachmentRetention';

describe('chatAttachmentRetention', () => {
  const now = new Date('2026-07-01T00:00:00.000Z');

  it('uses a 180 day retention window', () => {
    expect(CHAT_ATTACHMENT_RETENTION_DAYS).toBe(180);
  });

  it('expires attachments for old chat messages', () => {
    expect(isChatAttachmentExpired(new Date('2026-01-01T00:00:00.000Z'), now)).toBe(true);
    expect(isChatAttachmentExpired(new Date('2026-06-01T00:00:00.000Z'), now)).toBe(false);
  });

  it('accepts Firestore timestamp-like values', () => {
    const timestamp = {
      toDate: () => new Date('2026-01-01T00:00:00.000Z'),
    };

    expect(isChatAttachmentExpired(timestamp, now)).toBe(true);
  });
});
